app/views/AnalyticalFramework/PrimaryTagging/index.tsx

```diff
@@ -127,8 +128,14 @@ function PrimaryTaggingInput<K extends string>(props: PrimaryTaggingInput<K>) {
         (value: Section[]) => {
             setTempSections(undefined);
             setSections(value, name);
+            if (value?.length > 0 && value.find((tab) => tab.clientId === sectionToEdit)) {
+                setSelectedSection(sectionToEdit);
+            } else {
+                const firstTab = value?.length > 0 ? value[0].clientId : undefined;
+                setSelectedSection(firstTab);
+            }
```

Comment:
Are we sure value has at least one element?

Reply:
Yes, the sections element will have at least one element by default.

Reply:
Let's change this logic so that we check
1. value.length is less than zero then set undefined as selected section
2. value's clientId is equal to sectionToEdit then set that as selected section
3. set first element as selected section